### PR TITLE
Added tracking of Undertow Xnio metrics

### PR DESCRIPTION
**Pull request by tandrup**
Adds these metrics:
```
# HELP wildfly_xnio_io_thread_count IoThreadCount (org.xnio<type=Xnio, provider="nio", worker="management"><>IoThreadCount)
# TYPE wildfly_xnio_io_thread_count untyped
wildfly_xnio_io_thread_count{worker="\"management\"",} 2.0
wildfly_xnio_io_thread_count{worker="\"default\"",} 8.0
# HELP wildfly_xnio_core_worker_pool_size CoreWorkerPoolSize (org.xnio<type=Xnio, provider="nio", worker="management"><>CoreWorkerPoolSize)
# TYPE wildfly_xnio_core_worker_pool_size untyped
wildfly_xnio_core_worker_pool_size{worker="\"management\"",} 5.0
wildfly_xnio_core_worker_pool_size{worker="\"default\"",} 4.0
# HELP wildfly_xnio_max_worker_pool_size MaxWorkerPoolSize (org.xnio<type=Xnio, provider="nio", worker="management"><>MaxWorkerPoolSize)
# TYPE wildfly_xnio_max_worker_pool_size untyped
wildfly_xnio_max_worker_pool_size{worker="\"management\"",} 10.0
wildfly_xnio_max_worker_pool_size{worker="\"default\"",} 64.0
# HELP wildfly_xnio_worker_queue_size WorkerQueueSize (org.xnio<type=Xnio, provider="nio", worker="management"><>WorkerQueueSize)
# TYPE wildfly_xnio_worker_queue_size untyped
wildfly_xnio_worker_queue_size{worker="\"management\"",} 0.0
wildfly_xnio_worker_queue_size{worker="\"default\"",} 0.0
# HELP wildfly_xnio_connection_count ConnectionCount (org.xnio<type=Xnio, provider="nio", worker="management", address="/0.0.0.0:9990"><>ConnectionCount)
# TYPE wildfly_xnio_connection_count untyped
wildfly_xnio_connection_count{worker="\"management\", address=\"/0.0.0.0:9990\"",} -1.0
wildfly_xnio_connection_count{worker="\"default\", address=\"/0.0.0.0:8080\"",} -1.0
wildfly_xnio_connection_count{worker="\"default\", address=\"/0.0.0.0:8443\"",} -1.0
```

### CODE REVIEW

File: example_configs/wildfly-10.yaml [hunk 1]
```diff
@@ -25,3 +25,15 @@ rules:
       server: $1
       http_listener: $2
 
+  - pattern: "^org.xnio<type=Xnio, provider=\"nio\", worker=(.+)><>(.+_size|.+_count):"
+    attrNameSnakeCase: true
+    name: wildfly_xnio_$2
```
Comment (brian-brazil) [inline example_configs/wildfly-10.yaml]: Is xnio actually part of wildfly? It seems to be a separate project.

File: example_configs/wildfly-10.yaml [hunk 2]
```diff
@@ -25,3 +25,15 @@ rules:
...
+    labels:
+      worker: $1
+
+  - pattern: "^org.xnio<type=Xnio, provider=\"nio\", worker=(.+), address=(.+)><>(.+_count):"
```
Comment (brian-brazil) [inline example_configs/wildfly-10.yaml]: These are gauges, so you should remove the _count suffix

File: example_configs/wildfly-10.yaml (hunk 1 above)
Comment (tandrup) [reply example_configs/wildfly-10.yaml]: Xnio is a jboss project that ships as part of undertow. Undertow does not expose all metrics about worker threads but leaves that to xnio.

Comment (brian-brazil) [reply example_configs/wildfly-10.yaml]: If it's not part of the wildfly component, it doesn't get a wildfly prefix as it is its own thing.

**Comment by brian-brazil** [comment]
This has gone stale, so closing for now.

**Pull request closed without merging**